pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -90,7 +90,7 @@ func (c *RebalancerController) Reconcile(ctx context.Context, req controllerrunt
 		return controllerruntime.Result{}, err
 	}
 
-	if rebalancer.Status.FinishTime == nil {
+	if newStatus.FinishTime == nil {
 		// should never reach here.
```

Comment:
My question is why we need to keep this check since it `should never reach here`?

Reply:
It would be better to move this where referencing the `FinishTime`.